scipy/optimize/_differentialevolution.py

```diff
@@ -958,14 +958,15 @@ def _calculate_population_energies(self, population):
         num_members = np.size(population, 0)
         nfevs = min(num_members,
                     self.maxfun - num_members)
```

Comment:
I think there is an existing bug here. I think the line should be:
```
nfevs = min(num_members, self.maxfun - self._nfev)
```
Which means the line you added isn't needed.

Reply:
Should I change it this way?

Reply:
No thank you. Just remove the line you added, I'll make a separate PR.

Reply:
The `max` is needed for the `if calc_energies.size != nfevs:` to work, so I cannot remove it.

Reply:
I fixed this bug in #14520, so the line `nfevs = max(nfevs, 0)` is not needed.